src/server.h

```diff
@@ -255,6 +267,38 @@ inline bool Server<IoType, SwitchActivityInfo, SwitchCalcGaps>::server_receive_t
         hexdump(m_pMsgReply->getBuf(), MsgHeader::EFFECTIVE_SIZE);
 #endif /* LOG_TRACE_MSG_IN */
 
+        /* if requested persist each message to file */
+        if (s_user_params.b_write_msg_to_file) {
+
+            uint8_t *msg = m_pMsgReply->getBuf();
+            int msg_len = m_pMsgReply->getLength();
+
+            if (write_msg_file) {
+                assert(s_user_params.write_msg_file_flags_opt == DEFAULT_FILE_FLAGS_OPT);
+
+                size_t wb = fwrite(msg, sizeof(uint8_t), msg_len, write_msg_file);
+
+                if (wb != (size_t)msg_len) {
+                    log_err("Failed to write complete message to file stream - wrote %zu of %d bytes.", wb, msg_len);
+                    exit_with_log("Failed to write message to file stream.", SOCKPERF_ERR_FATAL);
+                }
+
+                if (fflush(write_msg_file) != 0) {
+                    exit_with_log("Failed to flush file stream.", SOCKPERF_ERR_FATAL);
+                }
+            } else {
+                assert(s_user_params.write_msg_file_flags_opt != DEFAULT_FILE_FLAGS_OPT);
+
+                memcpy(write_msg_buf_aligned, msg, msg_len);
+                ssize_t wb = write(write_msg_fd, write_msg_buf_aligned, msg_len);
+
+                if (wb != msg_len) {
```

Comment:
For consistency, cast `msg_len` to `size_t` as done in `if (write_msg_file){ ... }` above.